### ISSUE TITLE
Feature request: Parsing style that saves all unrecognized flags

### ISSUE THREAD

**Issue by kevinkassimo** [rust-lang/getopts#78]
Hi. So currently it seems that `getopts` does not support a parsing style that tries to match as many flags as possible, while saving the ones that it does not recognize to allow later user inspection. (such parsing style would be useful to mix up two different types of flags, one type which the user controls and the other type forwarded to a subprocess, etc.)

An example scenario: Node.js extracts flags it recognizes, which forwarding the remaining ones to the underlying v8 JavaScript engine.

It is possible to support this parsing style? Or is there some alternative way that this could be achieved?

**Comment by ExpHP** [comment]
This is more or less impossible. Say we have the following:

```rust
opts.optopt("n", "", "number of points", "COUNT")
```
and now suppose we want to forward unknown options to something that has a `--thing-flags` option:
```
prog --thing-flags -n 3
```
There are two distinct possibilities:
* `--thing-flags` takes no argument, and so your program should receive an `-n` flag with `COUNT=3`.
* `--thing-flags` takes an argument (the `-n`), and so your program should instead get `3` as a positional (`free`) argument.

In other words, **properly extracting the arguments to your own program requires knowledge of all of the flags of the thing you are wrapping.**
